Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/XDRPolicyTransformHelperTest.java

```diff
@@ -0,0 +1,219 @@
+
+		XDREventType eventA = new XDREventType();
+
+		assertionType.setMessageId(MESSAGE_ID);
+
+		slotType1.setName("$XDSDocumentEntryPatientId");
+
+		valueListType.getValue().add("1111.4444^^^&amp;26.489.22&amp;ISO");
+		slotType1.setValueList(valueListType);
+
+		pnt.setFullName("Minh");
+		assertionType.setPersonName(pnt);
+
+		rsl.getSlot().add(slotType1);
+
+		sor.setId("100");
+		sor.setRequestSlotList(rsl);
+
+		ProvideAndRegisterDocumentSetRequestType prdsrt = new ProvideAndRegisterDocumentSetRequestType();
+		Document doc = new Document();
+		doc.setId("223");
+		prdsrt.getDocument().add(doc);
+		prdsrt.setSubmitObjectsRequest(sor);
+
+		sor.setRegistryObjectList(rolt);
+
+		XDRMessageType xmt = new XDRMessageType();
+		eventA.setMessage(xmt);
+		xmt.setAssertion(assertionType);
+		xmt.setProvideAndRegisterDocumentSetRequest(prdsrt);
+
+		RequestType request = new RequestType();
+		SubjectHelper subjHelp = new SubjectHelper();
+		SubjectType subject = subjHelp.subjectFactory(hct, assertionType);
+		request.getSubject().add(subject);
+
+		assertionType.setHomeCommunity(hct);
+		assertionType.setSSN("170112756");
+		eventA.setSendingHomeCommunity(hct);
+		String direction = "inbound";
+		eventA.setDirection(direction);
+		eventA.getMessage().setAssertion(assertionType);
+
+		CheckPolicyRequestType result = xdrTransformHelper.transformXDRToCheckPolicy(eventA);
+
+		verifycheckPolicy_Request(result, direction);
+
+	}
+
+	@Test
+	public void testTransformXDRResponseToCheckPolicy() throws Exception {
+
+		XDRResponseEventType eventA = new XDRResponseEventType();
+
+		assertionType.setMessageId(MESSAGE_ID);
+
+		slotType1.setName("$XDSDocumentEntryPatientId");
+
+		valueListType.getValue().add("1111.4444^^^&amp;26.489.22&amp;ISO");
+		slotType1.setValueList(valueListType);
+
+		pnt.setFullName("Minh");
+		assertionType.setPersonName(pnt);
+
+		rsl.getSlot().add(slotType1);
+
+		sor.setId("100");
+		sor.setRequestSlotList(rsl);
+
+		RegistryResponseType prdsrt = new RegistryResponseType();
+
+		prdsrt.setResponseSlotList(rsl);
+		prdsrt.setRequestId("1235");
+
+		sor.setRegistryObjectList(rolt);
+
+		XDRResponseMessageType xmt = new XDRResponseMessageType();
+		eventA.setMessage(xmt);
+		xmt.setAssertion(assertionType);
+		xmt.setRegistryResponse(prdsrt);
+
+		assertionType.setHomeCommunity(hct);
+		assertionType.setSSN("170112756");
+		eventA.setSendingHomeCommunity(hct);
+		String direction = "inbound";
+		eventA.setDirection(direction);
+		eventA.getMessage().setAssertion(assertionType);
+
+		RequestType request = new RequestType();
+		SubjectHelper subjHelp = new SubjectHelper();
+		SubjectType subject = subjHelp.subjectFactory(hct, assertionType);
+		request.getSubject().add(subject);
+		CheckPolicyRequestType result = xdrTransformHelper.transformXDRResponseToCheckPolicy(eventA);
+
+		verifycheckPolicy_Request(result, direction);
+
+	}
+
+	private void verifycheckPolicy_Request(CheckPolicyRequestType actualCheckPolicy, String direction) {
+
```

Comment:
input parameter direction is not getting used anywhere in verifycheckPolicy_Request method.